buffer/src/test/java/io/netty/buffer/ReadOnlyDirectByteBufferBufTest.java

```diff
@@ -293,12 +292,12 @@ public void testWrapMemoryMapped() throws Exception {
         ByteBuf b2 = null;
 
         try {
-            output = new FileOutputStream(file).getChannel();
+            output = new RandomAccessFile(file, "rw").getChannel();
```

Comment:
This looks like it leaks, since you need to close `RandomAccessFile`. The documentation for `getChannel()` does not say closing the channel closes the `RandomAccessFile`. The documentation for `RandomAccessFile.close()` notes the opposite, that closing the file closes the channel. Looking at the source, it seems the channel holds an extra ref, so closing it does not close the `fd` held by `RandomAccessFile`.

Reply:
@ejona86 maybe I am missing something but we call `output.close()` in the finally block so it should be fine ?

Reply:
At risk of repeating myself: `output` is the _channel_. But you need to close the `RandomAccessFile`. It is subtle.

Reply:
@ejona86 sorry I missed this... This is not correct, closing the `FileChannel` here will also close the `RandomAccessFile` from which it was created. See also http://hg.openjdk.java.net/jdk8u/jdk8u-dev/jdk/file/4a1e42601d61/src/share/classes/sun/nio/ch/FileChannelImpl.java#l138

Reply:
Hmm... the decrement for the ref is also in RandomAccessFile, so the channel _must_ close the file. Okay. Very well. I continue to be amazed at how poor some documentation is to define resource management...